packages/sdk-auth/src/auth.js

```diff
@@ -216,21 +230,26 @@ export default class SdkAuth {
 
   anonymousFlow(anonymousId: string = '', config: CustomAuthOptions = {}) {
     const _config = this._getRequestConfig(config)
-    const request = SdkAuth._buildRequest(
+    let request = SdkAuth._buildRequest(
       _config,
       SdkAuth._enrichUriWithProjectKey(
         this.ANONYMOUS_FLOW_URI,
         _config.projectKey
       )
     )
 
-    if (anonymousId) request.body += `&anonymous_id=${anonymousId}`
+    if (anonymousId)
+      request = SdkAuth._appendToRequestBody(request, {
```

Comment:
I guess we could technically leave `request` as a constant and return here instead of re-assigning and returning below. Just a matter of flavor tho.

Reply:
Yes, didn't want to change such things much in this PR. Generally the composition in this code can be improve. Ideally, the `AuthRequeust` could have a `map` and `value` function. Then you could just `request.map(SdkAuth._appendToRequestBody(...))` which gives you a mapped request and then `value()` when you pass it somewhere. That's another story though.